File: panda-ij-plugin/src/eu/nyerel/panda/ijplugin/runner/calltree/model/EmptyCallTreeModel.java
package eu.nyerel.panda.ijplugin.runner.calltree.model;

import com.intellij.ui.treeStructure.treetable.ListTreeTableModelOnColumns;
import com.intellij.ui.treeStructure.treetable.TreeColumnInfo;
import com.intellij.util.ui.ColumnInfo;

import javax.swing.tree.DefaultMutableTreeNode;

/**
 * @author Rastislav Papp (rastislav.papp@gmail.com)
 */
public class EmptyCallTreeModel extends ListTreeTableModelOnColumns {

    private static final ColumnInfo[] COLUMNS = new ColumnInfo[]{
            new TreeColumnInfo("")
    };

    public EmptyCallTreeModel() {
        super(new DefaultMutableTreeNode("No monitoring data to display (hit 'Refresh' button)"), COLUMNS);
    }

}

File: panda-ij-plugin/src/eu/nyerel/panda/ijplugin/runner/calltree/model/EmptyCallTreeModel.kt
package eu.nyerel.panda.ijplugin.runner.calltree.model

import com.intellij.ui.treeStructure.treetable.ListTreeTableModelOnColumns
import com.intellij.ui.treeStructure.treetable.TreeColumnInfo
import com.intellij.util.ui.ColumnInfo

import javax.swing.tree.DefaultMutableTreeNode

/**
 * @author Rastislav Papp (rastislav.papp@gmail.com)
 */
class EmptyCallTreeModel : ListTreeTableModelOnColumns(DefaultMutableTreeNode("No monitoring data to display (hit 'Refresh' button)"), COLUMNS) {
    companion object {

        private val COLUMNS = arrayOf<ColumnInfo<*, *>>(TreeColumnInfo(""))
    }

}
